New device_platform: serial (supports Teddy Ruxpin)

renekliment opened the pull request:
**Description:**
This can be used for device that uses Arduino for example - like the Teddy Ruxpin project Tedlexa.

**Checklist:**
  - [x] Update CHANGELOG.md
  - [ ] Update Wiki (new platform, installation, configuration changes etc)
  - [x] PR Location (usually `master` for hotfixes, `dev` for general development)
  - [x] need to rename platform files to somethingplatform.py, so this can be renamed to serialplatform.py and it's not serial.py which breaks imports
  - [x] uses unmerged PRs - triggering, logging, therefor fails tests, rebase after that is merged
  - [x] hasn't been tested at all - do testing

@packetrat We'll support bears by default, so you don't have to worry about keeping the code compatible as it gets updated!

ImEmJay commented:
I see I'm not the only one who really liked it. And good that I didn't had
the time to start implementing the serial platform for him today.

René Kliment <notifications@github.com> schrieb am Fr., 6. Jan. 2017, 19:51:

> *Description:*
> This can be used for device that use Arduino for example - like the Teddy
> Ruxpin project Tedlexa.
>
> *DOESN'T WORK YET*
>
> *Checklist:*
>
>    - Update CHANGELOG.md
>    - Update Wiki (new platform, installation, configuration changes etc)
>    - PR Location (usually master for hotfixes, dev for general
>    development)
>    - need to rename platform files to somethingplatform.py, so this can
>    be renamed to serialplatform.py and it's not serial.py which breaks imports
>    - uses unmerged PRs - triggering, logging, therefor fails tests,
>    rebase after that is merged
>    - hasn't been tested at all - do testing
>
> @packetrat <https://github.com/packetrat> We'll support bears by default,
> so you don't have to worry about keeping the code compatible as it gets
> updated!
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/alexa-pi/AlexaPi/pull/134
> Commit Summary
>
>    - initial revision, doesn't work yet; defaults for Teddy Ruxpin
>
> File Changes
>
>    - *A* src/alexapi/device_platforms/serial.py
>    <https://github.com/alexa-pi/AlexaPi/pull/134/files#diff-0> (88)
>    - *M* src/config.template.yaml
>    <https://github.com/alexa-pi/AlexaPi/pull/134/files#diff-1> (14)
>    - *M* src/dev-requirements.txt
>    <https://github.com/alexa-pi/AlexaPi/pull/134/files#diff-2> (3)
>    - *A* src/scripts/inc/serial.sh
>    <https://github.com/alexa-pi/AlexaPi/pull/134/files#diff-3> (9)
>    - *M* src/scripts/setup.sh
>    <https://github.com/alexa-pi/AlexaPi/pull/134/files#diff-4> (1)
>
> Patch Links:
>
>    - https://github.com/alexa-pi/AlexaPi/pull/134.patch
>    - https://github.com/alexa-pi/AlexaPi/pull/134.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/alexa-pi/AlexaPi/pull/134>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAly4qND0DUpluM4NQc7lcqJczKDCOskks5rPo0zgaJpZM4Lc_p3>
> .
>

lechat commented on src/dev-requirements.txt:
```diff
@@ -1,3 +1,4 @@
 fakeRPiGPIO
 -e git+https://github.com/xtacocorex/CHIP_IO.git#egg=CHIP_IO
-pylint
+pyserial
```
This should be in `requirements.txt`

renekliment replied:
This is for CI. For users, it gets installed in setup, if they select proper device.

renekliment commented:
Rebased, fixed and tested with a simple Arduino code (will post later, probably a new alexa-pi organization repo).

@alexa-pi/team Please have a look, this should be easy.

@packetrat Please test. We will roll out a new version soon and it would be awesome, if we could guarantee 100% bear compatibility :-)

renekliment commented:
The example piece of code can be found here: https://github.com/alexa-pi/AlexaPi-Arduino Contributions are welcome there - it's also an open project. It's main purpose is however the testing of this serial platform and as a pointer for beginners.

renekliment commented:
14-day PR review window rule ... merging.

@packetrat Please test the `dev` branch now to make sure the bear is supported in the new version.

@alexa-pi/team Only 1 more PR to go to _1.4_. Please have a look at it (#169), so we don't have to wait until Friday for the 14 days to run out. Thank you!
